### DISCUSSION TITLE
Routers at api level, instead of at server level?

### DISCUSSION THREAD

**Discussion by evertdespiegeleer** [ecyrbe/zodios#365] [Q&A]
Hi all!

I was wondering whether or not it's possible to define routers (or something similar) at api level, rather than at server level.
In my current codebase, I define a set of apis using `makeApi`. I then import all these apis in my server code, and assign them each a route using `use` (check example code).

```ts
const helloRouter = zodiosRouter(helloApi)
const userRouter = zodiosRouter(userApi)

helloRouter.get('/', (req, res) => {
  res.json({
    greeting: `Hello ${req.query.name}`
  })
})

...

const app = zodiosApp()

app.use('/hello', helloRouter)
app.use('/user', userRouter)
```

This way of working is fine at server level, but afaik this implies that multiple clients are required to talk to the different apis/routes, as the client code has no knowledge of the server code. Client code typically only imports the api definitions, and these routes don't exist at this level. Hence, at client level, I do something like this:

```ts
export const helloApi = new Zodios('http://example.com/hello', helloApi)
export const userApi = new Zodios('http://example.com/user', userApi)
```

Hence, I'm essentially defining these routes twice. This example is pretty simple, but `use` can be used to branch routes in a much more complex manner. Having to define them twice is a little undesirable 😊

Can these routes be defined at api level?

Thanks for all the good work! Loving Zodios so far!

Evert

**Comment by ecyrbe** [comment]
Hello, thank you for the feedback.

When you say routes defined at the api level, do you mean something like this ?

Server:
```ts
const api = mergeApi({
  '/hello': helloApi,
  '/user': userApi,
});
```
Client declared with all routers:
```ts
const client = new Zodios('http://example.com', api);
```

**Reply by evertdespiegeleer** [reply]
Thanks for your quick reaction!
That's indeed what I'm implying. This isn't yet possible, right? It's not too hard to work around it, but it'd be a nice feature.

**Reply by ecyrbe** [reply]
Actually it's almost already possible.
I can expose this helper easilly.
Opening a feature request

**Reply by ecyrbe** [reply]
Done and available in `v10.8.0` of zodios core : https://www.zodios.org/docs/api/helpers#mergeapis

**Reply by evertdespiegeleer** [reply]
Awesome new feature, thanks!
